### ISSUE TITLE
Implement `TIME_STYLE="+DATE" ls`

### ISSUE THREAD

**Issue by sylvestre** [uutils/coreutils#3625]
We already support TIME_STYLE as an environment variable but only with the 4 different values.
GNU allows the use of +date

For example:
```
$ touch exe
$ TIME_STYLE="+%Y__%S" ls exe -al
-rw-r--r-- 1 sylvestre sylvestre 0 2022__58 exe
$ TIME_STYLE="+%Y-%m-%d %H:%M:%S.%N %z" ls exe -al
-rw-r--r-- 1 sylvestre sylvestre 0 2022-06-13 21:34:58.728327312 +0200 exe
$ TIME_STYLE="+foobar" ls exe -al
-rw-r--r-- 1 sylvestre sylvestre 0 foobar exe

```

GNU supports the date format (like the date or touch commands)
https://github.com/uutils/coreutils/blob/main/src/uu/date/src/date.rs
https://github.com/uutils/coreutils/blob/main/src/uu/touch/src/touch.rs

Tests in:
https://github.com/coreutils/coreutils/blob/master/tests/ls/color-term.sh#L23=

**Comment by dmatos2012** [comment]
Hi. I would also like to try this issue. I have made some progress, and technically it works except for the clap validation, see problem on a question I made to [clap-repo](https://github.com/clap-rs/clap/discussions/4260#discussioncomment-3736434).

Basically right now we have [value_parser!](https://github.com/uutils/coreutils/blob/main/src/uu/ls/src/ls.rs#L1518), but not supporting +FORMAT. Thus, by supporting +FORMAT, it cannot match the string anymore, but rather a user defined string with the `+` key in front, thus making it a bit more difficult, and as far as I am concerned, requiring a custom logic in case parse fails.

In my question to clap, the maintainer suggested something, and I tried some stuff but it looks so far a bit beyond my capabilities, so if anyone has any other ideas, lets discuss :)

**Comment by tertsdiepraam** [comment]
Hi! Sounds very promising! You could always remove the parsing from clap, accept any string and then manually parse the string value later. What argument is this about by the way? In the issue I only see an environment variable, which clap should not have anything to do with right?

**Comment by dmatos2012** [comment]
Hi! Your initial suggestion seems like doable. The argument is [`--time-style`](https://github.com/uutils/coreutils/blob/b43bbe9e98d60b05cd98a02969d608cb1622f0d4/src/uu/ls/src/ls.rs#L1511-L1520) which accepts(for now)  `  .value_parser(["full-iso", "long-iso", "iso", "locale"])`. However if I add there `+FORMAT` then its no longer accepted. If I remove completely validation it works fine, but then you dont get the nice clap help message such as: `Possible values are: ["full-iso", "long-iso", "iso", "locale"] etc.

**Comment by tertsdiepraam** [comment]
Yeah we indeed lose the message and we'd need some custom message instead, but I'm not sure that the message from clap would still be as nice with a custom value_parser.
